Source/JavaScriptCore/dfg/DFGStrengthReductionPhase.cpp

```diff
@@ -1313,7 +1313,48 @@ class StrengthReductionPhase : public Phase {
                 break;
             }
 #endif
-            
+
+            // We gave up inlining a wrapped function, but still, we can inline bound function's wrapper by extracting it.
+            // This also wipes bound-function thunk call which is suboptimal compared to directly calling a wrapped function here.
+            if (executable->intrinsic() == BoundFunctionCallIntrinsic && function && (m_node->op() == Call || m_node->op() == TailCall || m_node->op() == TailCallInlinedCaller)) {
+                JSBoundFunction* boundFunction = jsCast<JSBoundFunction*>(function);
+                if (JSFunction* targetFunction = jsDynamicCast<JSFunction*>(boundFunction->targetFunction())) {
+                    auto* targetExecutable = targetFunction->executable();
+                    JSImmutableButterfly* boundArgs = boundFunction->boundArgs();
+                    if (((boundArgs ? boundArgs->length() : 0) + m_node->numChildren()) <= Options::maximumDirectCallStackSize()) {
+                        if (FunctionExecutable* functionExecutable = jsDynamicCast<FunctionExecutable*>(targetExecutable)) {
+                            // We need to update m_parameterSlots before we get to the backend, but we don't
+                            // want to do too much of this.
+                            unsigned numAllocatedArgs = static_cast<unsigned>(functionExecutable->parameterCount()) + 1;
+                            if (numAllocatedArgs <= Options::maximumDirectCallStackSize())
+                                m_graph.m_parameterSlots = std::max(m_graph.m_parameterSlots, Graph::parameterSlotsForArgCount(numAllocatedArgs));
+                        }
+
+                        unsigned firstChild = m_graph.m_varArgChildren.size();
+                        m_graph.m_varArgChildren.append(m_insertionSet.insertConstant(m_nodeIndex, m_node->origin, targetFunction)); // Callee.
```

Comment:
Let's use the same stack slot notation as below in this comment: `|callee|` instead of `Callee`.

Reply:
Fixed.